packages/services/api/src/modules/organization/resolvers.ts

```diff
@@ -1,13 +1,43 @@
 import { z } from 'zod';
 import { createConnection } from '../../shared/schema';
 import { AuthManager } from '../auth/providers/auth-manager';
+import { isOrganizationScope } from '../auth/providers/organization-access';
+import { isProjectScope } from '../auth/providers/project-access';
+import { isTargetScope } from '../auth/providers/target-access';
 import { IdTranslator } from '../shared/providers/id-translator';
 import { Logger } from '../shared/providers/logger';
 import type { OrganizationModule } from './__generated__/types';
 import { OrganizationManager } from './providers/organization-manager';
 
 const OrganizationNameModel = z.string().min(2).max(50);
 
+const builtInRole = z.custom<string>(val => {
+  if (val === 'Viewer' || val === 'Admin') {
+    return false;
+  }
+
+  return true;
+}, 'Viewer and Admin are reserved role names');
+
+const capitalLetter = z.custom<string>(val => {
+  return typeof val === 'string' && val.length > 0 && val[0] === val[0].toUpperCase();
+}, 'Role name must start with a capital letter');
+
+const createOrUpdateMemberRoleInputSchema = z.object({
+  name: z
+    .string({
+      required_error: 'Please enter role name',
+    })
+    .min(2, 'Role name must be at least 2 characters long')
+    .pipe(capitalLetter)
+    .pipe(builtInRole),
+  description: z
+    .string({
+      required_error: 'Please enter role description',
+    })
+    .min(2, 'Role description must be at least 2 characters long'),
```

Comment:
Also, a max length so nobody tries to store Harry Potter and the Order of Phoenix as the description.

Reply:
[a0941ac](https://github.com/kamilkisiela/graphql-hive/pull/3364/commits/a0941acff52342b33fc07bb5f65c43a87ac6480a)